### PR TITLE
feat: Add transformer to sanitize empty values

### PR DESCRIPTION
**Pull request by CrownKira**
## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

File: frontend/src/pages/onboarding/index.tsx [hunk 1]
```diff
@@ -47,15 +55,19 @@ const onboardingSteps: OnboardingStep[] = [
...
+        .required('Name is required.'),
     },
   },
   {
     title: 'Complete your profile',
     subtitle: 'Help other climbers know more about you',
     form: <DetailsForm />,
     validate: {
-      height: Yup.number().positive().integer().max(300).optional(),
+      height: Yup.number().positive().integer().max(MAX_HEIGHT).optional(),
```
Comment (Artemis-Hunt) [inline frontend/src/pages/onboarding/index.tsx:70]: Would be good to have a MIN_HEIGHT (BE is set at 100 I believe)

File: frontend/src/components/hook-form/RHFAutoMultiSelect.tsx [hunk 2]
```diff
@@ -54,14 +54,7 @@ export default function RHFAutoMultiSelect({ name, options = [], ...other }: Pro
...
           renderInput={(params) => (
-            <TextField
-              {...field}
-              {...params}
-              inputRef={ref}
-              error={!!error}
-              helperText={error?.message}
-              {...other}
-            />
+            <TextField {...params} error={!!error} helperText={error?.message} {...other} />
```
Comment (Artemis-Hunt) [inline frontend/src/components/hook-form/RHFAutoMultiSelect.tsx:57]: Just curious, what is the rationale behind removing this>

**Review by Artemis-Hunt — CHANGES_REQUESTED** [review]

File: frontend/src/pages/onboarding/index.tsx (hunk 1 above)
Comment (CrownKira) [reply frontend/src/pages/onboarding/index.tsx:70]: agree

File: frontend/src/components/hook-form/RHFAutoMultiSelect.tsx (hunk 2 above)
Comment (CrownKira) [reply frontend/src/components/hook-form/RHFAutoMultiSelect.tsx:57]: removed {...field} cos idw to let the RHF control the manage the input and value for this field component

Comment (Artemis-Hunt) [reply frontend/src/components/hook-form/RHFAutoMultiSelect.tsx:57]: I see

File: frontend/src/pages/onboarding/index.tsx [hunk 3]
```diff
@@ -47,15 +55,19 @@ const onboardingSteps: OnboardingStep[] = [
...
       reach: Yup.number().positive().integer().optional(),
```
Comment (Artemis-Hunt) [inline frontend/src/pages/onboarding/index.tsx:71]: Can consider changing Reach to be a selector from -20cm to +20cm (@therizhao confirmed that these are the lower and upper bounds for reach). Not sure if there's a better way to implement this

File: frontend/src/pages/onboarding/index.tsx [hunk 4]
```diff
@@ -119,7 +131,7 @@ export default function Onboarding() {
   const { handleSubmit, trigger, watch } = methods;
```
Comment (Artemis-Hunt) [inline frontend/src/pages/onboarding/index.tsx:132]: Would be good to have "Next" button become "Skip" when all fields in the form are optional AND there is no input (or empty input) for all the fields (For instance, on the Grades/Certification pages)

File: frontend/src/pages/onboarding/index.tsx [hunk 5]
```diff
@@ -119,7 +131,7 @@ export default function Onboarding() {
...
 
   // for debugging
-  useWatchForm(watch);
+  useDevWatchForm(watch);
```
Comment (Artemis-Hunt) [inline frontend/src/pages/onboarding/index.tsx:135]: The bouldering/top rope/lead climbing/SCNS grades are optional. Hence, the hint text should have (optional) e.g. Grade (optional). Also, it seems that FE sends some invalid data to BE when the blank option is selected (See the ClimbJios Staging Alerts tele channel)

File: frontend/src/pages/onboarding/index.tsx [hunk 6]
```diff
@@ -47,15 +55,19 @@ const onboardingSteps: OnboardingStep[] = [
     subtitle: 'Other climbers will use this to identify you',
```
Comment (Artemis-Hunt) [inline frontend/src/pages/onboarding/index.tsx:55]: Can consider center-aligning the subtitle text (such as 'Other Climbers will use this to identify you') for all onboarding pages

**Pull request merged**